pkg/armhelpers/deployer.go

```diff
@@ -0,0 +1,59 @@
+package armhelpers
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/arm/resources/resources"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/prometheus/common/log"
+)
+
+type AzureTemplateDeployer struct{}
+
+func (az *AzureClient) Validate(resourceGroup, name string, d resources.Deployment) (resources.DeploymentValidateResult, error) {
+	return az.DeploymentsClient.Validate(resourceGroup, name, d)
+}
+
+func (az *AzureClient) CreateOrUpdate(resourceGroup, name string, d resources.Deployment, cancel chan struct{}) (<-chan resources.DeploymentExtended, <-chan error) {
+	return az.DeploymentsClient.CreateOrUpdate(resourceGroup, name, d, cancel)
+}
+
+func (az *AzureClient) Delete(resourceGroup, name string, cancel chan struct{}) (<-chan autorest.Response, <-chan error) {
+	return az.DeploymentsClient.Delete(resourceGroup, name, cancel)
+}
+
+func (az *AzureClient) Get(resourceGroup, name string) (resources.DeploymentExtended, error) {
+	return az.DeploymentsClient.Get(resourceGroup, name)
+}
+
+func DeployTemplate(deployer TemplateDeployer, resourceGroupName, deploymentName string, template map[string]interface{}, parameters map[string]interface{}) (response *resources.DeploymentExtended, err error) {
```

Comment:
I might be missing the purpose of AzureTemplateDeployer. Shouldn't DeployTemplate be associated with AzureTemplateDeployer? Or it just supposed to take TemplateDeployer (which is fine as well) then shouldn't CreateOrUpdate be associated with AzureTemplateDeployer?